src/components/ListItem.jsx

```diff
@@ -1,5 +1,60 @@
+import { useState, useEffect } from 'react';
+
+import { updateItem } from '../api/firebase';
+
+import { getFutureDate } from '../utils';
+
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	listToken,
+	itemId,
+	name,
+	isChecked,
+	dateLastPurchased,
+}) {
+	const twentyFourHoursInMilleSeconds = 86400000;
+	// sync up checked or not checked data from the database to the page upon page refresh
+	const [isPurchased, setIsPurchased] = useState(isChecked);
+
+	useEffect(() => {
+		// only update item when the checkbox has been checked. Don't update when the page just loaded and "isPurchased" is changed accordingly
+		if (isChecked !== isPurchased) {
+			updateItem(listToken, {
+				itemId: itemId,
+				isChecked: isPurchased,
+				currentTime: getFutureDate(0),
+			});
+		}
+	}, [isPurchased, isChecked, itemId, listToken]);
+
+	useEffect(() => {
```

Comment:
Nice use of the useEffect hook here (and setup earlier) to check and update time from last purchase!